### PR TITLE
Cart an Wish components

### PR DESCRIPTION
**Pull request by 5P4C3GR33D**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (alexpassss) [inline .vscode/settings.json:1]: Add this file to the .gitignore

File: src/app/components/Lib/Table/TableHead.template.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { Container, Row, Col } from '@qonsoll/react-design'
+import {
+  TableHead,
+  TableRow,
+  TableCell,
+  TableSortLabel,
+  Checkbox
+} from '@material-ui/core/'
+
+const ROUTS = {
```
Comment (alexpassss) [inline src/app/components/Lib/Table/TableHead.template.js:14]: Mistake. ROUTES

File: src/app/components/Lib/Table/TableHead.template.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  cart: 'CART',
+  wishes: 'WISHES'
+}
+
+const useStyles = makeStyles({
```
Comment (alexpassss) [inline src/app/components/Lib/Table/TableHead.template.js:19]: Move to the separate file

File: src/app/components/Lib/Table/TableHead.template.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  TableHead,
+  TableRow,
+  TableCell,
+  TableSortLabel,
+  Checkbox
+} from '@material-ui/core/'
+
+const ROUTS = {
+  cart: 'CART',
```
Comment (alexpassss) [inline src/app/components/Lib/Table/TableHead.template.js:15]: Seems to be a useless object or incorrect interpretation

File: src/app/components/Lib/Table/TableHead.template.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types'
+import { React } from 'react'
+import { tableCells } from 'app/constants'
```
Comment (alexpassss) [inline src/app/components/Lib/Table/TableHead.template.js:3]: camelCase to UPPER_CASE

File: src/app/components/Lib/Table/TableHead.template.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    padding: '6px'
+  }
+})
+
+const CustomTableHead = (props) => {
+  const classes = useStyles(props)
+  const cells =
+    props.page === ROUTS.cart ? tableCells.CART_CELLS : tableCells.WISH_CELLS
+  return (
+    <Container display="box">
```
Comment (alexpassss) [inline src/app/components/Lib/Table/TableHead.template.js:32]: Useless display property

File: src/app/components/Lib/Table/TableHead.template.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+              ))}
+            </TableRow>
+          </TableHead>
+        </Col>
+      </Row>
+    </Container>
+  )
+}
+CustomTableHead.propTypes = {
+  page: PropTypes.oneOf(['WISHES', 'CART'])
```
Comment (alexpassss) [inline src/app/components/Lib/Table/TableHead.template.js:56]: Array to constants

File: src/app/components/Lib/Toolbar/Toolbar.template.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import PropTypes from 'prop-types'
+import CheckIcon from '@material-ui/icons/Check'
+import DeleteIcon from '@material-ui/icons/Delete'
+import { Container, Row, Col } from '@qonsoll/react-design'
+import { Button, Toolbar, IconButton, Typography } from '@material-ui/core/'
+
+const Tolbar = (props) => {
+  return (
+    <Container display="box">
```
Comment (alexpassss) [inline src/app/components/Lib/Toolbar/Toolbar.template.js:9]: Useless display property

File: src/app/components/Lib/Toolbar/Toolbar.template.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      <Row>
+        <Col>
+          <Toolbar>
+            {props.numberChecked > 0 ? (
```
Comment (alexpassss) [inline src/app/components/Lib/Toolbar/Toolbar.template.js:13]: numberChecked > 0 means that value should be 1, 2 or higher

1 == true

Useless comparison

File: src/app/components/Lib/Toolbar/Toolbar.template.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+              <Container display="box">
+                <Row h="right" v="center">
+                  <Col>
+                    <Typography variant="button">
+                      {props.numberChecked + ' selected'}
```
Comment (alexpassss) [inline src/app/components/Lib/Toolbar/Toolbar.template.js:18]: Calculate values separately from template section. Use multiline string

File: src/app/components/Lib/Toolbar/Toolbar.template.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+                </Row>
+              </Container>
+            )}
+          </Toolbar>
+        </Col>
+      </Row>
+    </Container>
+  )
+}
+Tolbar.propTypes = {
```
Comment (alexpassss) [inline src/app/components/Lib/Toolbar/Toolbar.template.js:47]: Mistake. Toolbar

File: src/app/components/Lib/Toolbar/Toolbar.template.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+  numberChecked: PropTypes.number.isRequired
+}
+Tolbar.defaultProps = {}
```
Comment (alexpassss) [inline src/app/components/Lib/Toolbar/Toolbar.template.js:50]: Remove defaultProps

File: src/app/domains/Cart/components/index.js [hunk 13]
```diff
@@ -0,0 +1,2 @@
+import { CartTable } from './table'
+export { CartTable }
```
Comment (alexpassss) [inline src/app/domains/Cart/components/index.js:2]: use default export

File: src/app/domains/Cart/components/table/CartTable.template.js [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+const rows = [
+  createData('Ruslan', 'Rostik', 'Sugar', 'Kitchen'),
+  createData('Ruslan', 'Lena', 'Waffles', 'Kitchen'),
+  createData('Ruslan', 'Rostik', 'Sugar', 'Kitchen'),
+  createData('Ruslan', 'Rostik', 'Sugar', 'Kitchen'),
+  createData('Ruslan', 'Rostik', 'Sugar', 'Kitchen'),
+  createData('Ruslan', 'Rostik', 'Sugar', 'Kitchen')
+]
+
+const useStyles = makeStyles({
```
Comment (alexpassss) [inline src/app/domains/Cart/components/table/CartTable.template.js:20]: Move styles to the separate file

**Review by alexpassss — CHANGES_REQUESTED** [review]

File: src/app/domains/Wish/components/table/Wish.template.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+import { Container, Row, Col } from '@qonsoll/react-design'
+
+const WishTable = (props) => {
+  const [numSelected, setNumSelected] = useState([])
+  return (
+    <Container>
+      <Row h="center">
+        <Col>
+          <Toolbar numSelected={numSelected} />
+          <DataGrid collection="wishes" setNumSelected={setNumSelected} />
```
Comment (VladBogach) [inline src/app/domains/Wish/components/table/Wish.template.js:13]: create constants for collection names

File: src/app/domains/Cart/components/table/Cart.template.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+import { Container, Row, Col } from '@qonsoll/react-design'
+
+const CartTable = (props) => {
+  const [numSelected, setNumSelected] = useState([])
+  return (
+    <Container>
+      <Row h="center">
+        <Col>
+          <Toolbar numSelected={numSelected} />
+          <DataGrid collection="cart" setNumSelected={setNumSelected} />
```
Comment (VladBogach) [inline src/app/domains/Cart/components/table/Cart.template.js:13]: create constants for collections names

File: src/app/components/Lib/Toolbar/Toolbar.template.js [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+    <Container>
+      <Row>
+        <Col>
+          <Toolbar disableGutters>
+            {props.numSelected ? (
+              <Container>
+                <Row h="right" v="center">
+                  <Col pl={2}>
+                    <Typography variant="button">
+                      {`${props.numSelected} selected`}
```
Comment (VladBogach) [inline src/app/components/Lib/Toolbar/Toolbar.template.js:18]: move to separate variable

File: src/app/components/Lib/DataGrid/DataGrid.template.js [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+    id: 5,
+    assignedUser: 'Lannister',
+    productCategory: 'Office',
+    productName: 'Pen'
+  }
+]
+
+const CustomDataGrid = (props) => {
+  const cells =
+    props.collection === 'cart'
```
Comment (VladBogach) [inline src/app/components/Lib/DataGrid/DataGrid.template.js:43]: create constants for collections names

File: src/app/domains/Cart/components/table/Cart.template.js [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+        </Col>
+      </Row>
+    </Container>
+  )
+}
+
+CartTable.propTypes = {
+  number: PropTypes.number.isRequired
```
Comment (VladBogach) [inline src/app/domains/Cart/components/table/Cart.template.js:21]: check this field

**Pull request closed without merging**